File: golang/35-search-insert-position.go
package golang

import (
	"math"
)

func searchInsert(nums []int, target int) int {
	left := 0
	right := len(nums) - 1
	for left < right {
		mid := int(math.Floor(float64((left + right) / 2)))
		if nums[mid] == target {
			return mid
		}
		if nums[mid] > target {
			right = mid
		} else {
			left = mid + 1
		}
	}
	if target <= nums[left] {
		return left
	} else {
		return left + 1
	}
}

File: typescript/35-search-insert-position.ts
function searchInsert(nums: number[], target: number): number {
    let left = 0
    let right = nums.length - 1
    while (left < right) {
        let mid = Math.floor((left+right)/2)
        if (nums[mid] === target) return mid
        if (nums[mid] > target) {
            right = mid
        } else {
            left = mid + 1
        }
    }
    return target <= nums[left] ? left : left + 1
}